### PR TITLE
docs: release notes for v0.12.0

### PR DESCRIPTION
**Pull request by jm-clius**
The proposed release notes for [Release v0.12.0](https://github.com/status-im/nwaku/milestone/4).

Once the last open PRs are merged, I'll rebase this on the release candidate on `master`.

For the items you were responsible for, please check:

- that it's included in the release notes
- that you're happy with the phrasing

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,72 @@
...
+- Github CI test execution now skipped for doc-only changes.
+
+### Fixes
+
+- Fixed nwaku unnecessary sleep when no dynamic bootstrap nodes retrieved.
+- Fixed [`12/WAKU2-FILTER`](https://rfc.vac.dev/spec/12/) not working from browser-based clients due to nwaku peer manager failing to reuse existing connection.
+- Waku Message payload now correctly encoded as base64 in the Relay REST API.
+- Fixed handling of bindParam(uint32) in sqlite.
+- `chat2` application now correctly selects a random store node on startup.
+- Fixed macos builds failing due to unsupported dependency.
```
Comment (rymnc) [inline CHANGELOG.md:48]:
```suggestion
- Fixed macos builds failing due to an unsupported dependency.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,72 @@
...
+- The performance and stability of the message `store` has improved dramatically. Query durations, even for long-term stores, have improved by more than a factor of 10.
+- Support for Waku Peer Exchange - a discovery method for resource-restricted nodes.
+- Messages can now be marked as "ephemeral" to prevent them from being stored.
+- [Zerokit](https://github.com/vacp2p/zerokit) is now the default implementation for spam-protected `relay` with RLN.
+
+The full list of changes is below.
+
+### Features
+
+- Default support for [Zerokit](https://github.com/vacp2p/zerokit) version of [`17/WAKU-RLN-RELAY`](https://rfc.vac.dev/spec/17/) implementation.
```
Comment (rymnc) [inline CHANGELOG.md:13]:
```suggestion
- Default support for [Zerokit](https://github.com/vacp2p/zerokit) version of [`17/WAKU2-RLN-RELAY`](https://rfc.vac.dev/spec/17/) implementation.
```
The naming is fixed in an unmerged PR on the RFC repo, we can leave it as is in the release notes if you'd like!

**Review by rymnc — APPROVED** [review]
LGTM, thanks 🥳

**Review by LNSD — APPROVED** [review]
LGTM ✅

**Pull request merged**